### DISCUSSION TITLE
Memory: Raycast throwing out of bounds.

### DISCUSSION THREAD

**Discussion by DennisSmolek** [jrouwe/JoltPhysics.js#150] [Q&A] [answered]
Similar to #149 I am stress testing my application loading and unloading things. 
I am properly destroying things, and although difficult to profile, it seems I'm not breaking my memory limits.
However, at somepoint firing a ray triggers an out of bounds error. 
(Which is crazy because I have another demo firing 60,000 rays a second)
![image](https://github.com/jrouwe/JoltPhysics.js/assets/1397052/4d258871-5d29-4949-9161-6aaf9abf7a33)

This is the more detailed stack.

The raycaster casts twice before this no problem, in this case it did get a new collector prior to firing, but the setting of the collector tests successfully.

Is it likely the raycasting is causing the error or is this just the request that finally went out of bounds?

I have the tooling setup to debug this better (can walk through the c++ codebase), and we can probably get this live onto a public page, however it's wrapped in react.

**Comment by jrouwe** [comment, marked as answer]
So it seems to be crashing in a memory reallocation to add an additional hit to the collector. As with #149 it points to general memory corruption and I have the same suggestion as I provided in #149.

**Comment by DennisSmolek** [comment]
So what I've learned is it's likely never a true out of bounds situation but I totally broke something along the way.
Your comment about memory corruption had me go through earlier calls/stacks and after some old school logging/testing I figured out I was over eagerly destroying things that turns out where needed. 
I need to spend time getting the profiler working like you mentioned in #149
